authors/apps/articles/views.py

```diff
@@ -275,3 +275,80 @@ def get(self, request):
         return Response(
             {'tags': msg},
             status=status.HTTP_404_NOT_FOUND)
+
+
+class BookmarkAPIView(generics.ListAPIView):
+    """
+    Class returns all articles in bookmarks of the user logged in
+    """
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.BookmarkSerializer
+    renderer_classes = (BookmarkJSONRenderer,)
+
+    def get_queryset(self):
+        """
+        This view should return a list of all the bookmarks
+        for the currently authenticated user.
+        """
+        user = self.request.user
+        return Bookmark.objects.filter(user=user)
+
+
+class ArticleBookmarkAPIView(generics.GenericAPIView):
+    """Class adds or deletes an article from users bookmarks"""
+    permission_classes = (permissions.IsAuthenticated,)
+    serializer_class = serializers.BookmarkSerializer
+
+    def post(self, request, slug):
+        """
+        Add article to bookmarks of the user
+        Checks to see article being added exists
+        Checks to see if article has already been added to boomarks
+        """
+        article = get_single_article_using_slug(slug)
+        if not article:
+            error = f'An article with this slug,{slug}, does not exist'
+            return Response({
+                'error': error,
+                'status': status.HTTP_404_NOT_FOUND},
+                status=status.HTTP_404_NOT_FOUND)
+        try:
```

Comment:
Thanks for the good work @missvicki. But don't you think this method for querying the database for a bookmark should be in model helpers as someone using our app in the future might want it and they will have to write the same in their functionalities which is a duplication of code.
I'd rather you put it in model helpers.
I think this is mixing concerns. Model helpers shouldn't be in Views.
Thanks again for the Good work

Reply:
this has been resolved